File: js/day-2/part-2.js
const { read } = require("../utils/read");

(async () => {
  let depth = 0;
  let horizontal = 0;
  let aim = 0;

  const data = await read(`${__dirname}/data`);

  for (let i = 0; i < data.length; i++) {
    const datum = data[i];

    let [action, number] = datum.split(" ");
    number = Number(number);

    if (action === "forward") {
      horizontal += number;
      depth += aim * number;
    }

    if (action === "down") {
      aim += number;
    }

    if (action === "up") {
      aim -= number;
    }
  }

  console.log(`result: ${horizontal * depth}`);
})();

File: go/day-2/part-2.go
package main

import (
	"fmt"
	"os"
	"strconv"
	"strings"

	"aoc-2021/internals/utils"
)

func main() {
    data, err := os.ReadFile("./data")
    utils.Check(err)
	
	lines := strings.Split(string(data), "\n")

	var depth int64 = 0
	var horizontal int64 = 0
	var aim int64 = 0

	for i := 0; i < len(lines); i++ {
		line := lines[i]
		split := strings.Split(line, " ")
		
		action := split[0]

		number, err := strconv.ParseInt(split[1], 10, 64)
		utils.Check(err)

		if (action == "forward") {
			horizontal += number;
			depth += aim * number;
		}

		if (action == "down") {
			aim += number;
		}

		if (action == "up") {
			aim -= number;
		}
	}

	fmt.Print("result: ", horizontal * depth)
}